imageio/plugins/opencv.py

```diff
@@ -0,0 +1,219 @@
+            provided ndimage is a single image. If the provided ndimage is a list of images,
+            this parameter has no effect.
+        params : List[int]
+            A list of parameters that will be passed to OpenCVs imwrite or
+            imwritemulti functions. Possible values are documented in the
+            `OpenCV documentation
+            <https://docs.opencv.org/4.x/d4/da8/group__imgcodecs.html#gabbc7ef1aa2edfaa87772f1202d67e0ce>`_.
+
+        Returns
+        -------
+        encoded_image : bytes, None
+            If the ImageResource is ``"<bytes>"`` the call to write returns the
+            encoded image as a bytes string. Otherwise it returns None.
+
+        """
+
+        if isinstance(ndimage, list):
+            ndimage = np.stack(ndimage, axis=0)
+        elif not is_batch:
+            ndimage = ndimage[None, ...]
+
+        if ndimage[0].ndim == 2:
+            n_channels = 1
+        else:
+            n_channels = ndimage[0].shape[-1]
+
+        if n_channels == 1:
+            ndimage_cv2 = [x for x in ndimage]
+        elif n_channels == 4:
+            ndimage_cv2 = [cv2.cvtColor(x, cv2.COLOR_RGBA2BGRA) for x in ndimage]
+        else:
+            ndimage_cv2 = [cv2.cvtColor(x, cv2.COLOR_RGB2BGR) for x in ndimage]
+
+        retval = cv2.imwritemulti(self.file_handle, ndimage_cv2, params)
+
+        if retval is False:
+            # not sure what scenario would trigger this, but
+            # it can occur theoretically.
+            raise IOError("OpenCV failed to write.")  # pragma: no cover
+
+        if self.request._uri_type == URI_BYTES:
+            return Path(self.file_handle).read_bytes()
+
+    def properties(self, index: int = 0, **kwargs) -> ImageProperties:
+        """Standardized image metadata.
+
+        Parameters
+        ----------
+        index : int, Ellipsis
+            The index of the image to read metadata for. See ``read`` for details.
+            Defaults to 0.
+        kwargs
+            Additional kwargs are forwarded to OpenCVs ``imread`` or
+            ``imreadmulti`` function.
+
+        Returns
+        -------
+        props : ImageProperties
+            A dataclass filled with standardized image metadata.
+
+        Notes
+        -----
+        Reading properties with OpenCV involves decoding pixel data, because
+        OpenCV doesn't provide a direct way to access metadata.
+
+        """
+
+        # unfortunately, OpenCV doesn't allow reading shape without reading pixel data
+        img = self.read(index=index, **kwargs)
+
+        return ImageProperties(
+            shape=img.shape,
+            dtype=img.dtype,
+            is_batch=(index is ...),
+        )
+
+    def metadata(self, index: int = 0, exclude_applied: bool = True) -> Dict[str, Any]:
+        """Format-specific metadata.
+
+        .. warning::
+            OpenCV does not support reading metadata. When called, this function
+            will raise a ``NotImplementedError``.
```

Comment:
Or should it just return an empty dict then?

Reply:
That is the alternative, yes. I chose to `raise` here because an empty dict could also be interpreted as "there is no metadata".  We could do warning + empty dict as well if you prefer that though.

Reply:
I will make it raise a warning an return an empty dict. I think this will work out smoother for us down the line when we want to allow the v2 API to use v3 plugins (in which case we always have to read metadata).